lib/Zonemaster/Backend/ErrorMessages.pm

```diff
@@ -0,0 +1,96 @@
+package Zonemaster::Backend::ErrorMessages;
+
+
+use Readonly;
+
+# This structure is used to replace messages coming from JSON::Validator
+# The items are evaluated in order and the first item that have its `pattern`
+#  key matching the current path will be used.
+# The `config` key holds the custom messages, the keys of the first level hash
+#  represent the types of the JSON element and the ones of the second level
+#  hash are the types of error.
+# This structure is similar to the one used internally in JSON::Validator,
+#  see https://github.com/jhthorsen/json-validator/blob/master/lib/JSON/Validator/Error.pm
+
+Readonly my @CUSTOM_MESSAGES_CONFIG => (
+    {
+        pattern => "/(domain|hostname)",
+        config => {
+            string => {
+                pattern => 'The domain name character(s) are not supported'
```

Comment:
'The domain name contains a character or characters not supported'

Reply:
done